### PR TITLE
Desktop: Fix #4581: Show completed todos based on showCompletedTodos setting

### LINKED ISSUE #4581: Search result with tasks and notes
A search with a query like `notebook:notebook1` give as result, notes and tasks completed and not completed.

Tasks completed are also shown if the option “show completed to-dos” is unchecked.

The filter `iscompleted:0` is not useful because in this case notes are not visible.

## Environment

Joplin version: 1.7.11
Platform: Windows
OS specifics: Win10

## Steps to reproduce

1. Add notes and tasks checked and unchecked in your notebook1
2. Be sure the option “show completed to-dos” is unchecked (View menu)
3. Type this query in the search field : `notebook:notebook1`

## Describe what you expected to happen

Not show completed tasks when “show completed to-dos” is unchecked.

### PR DESCRIPTION
**Pull request by JackGruber**
When searching show the completed todos based on the showCompletedTodos setting.

### CODE REVIEW

File: packages/lib/services/searchengine/SearchEngineUtils.ts [hunk 1]
```diff
@@ -1,17 +1,23 @@
 import SearchEngine from './SearchEngine';
 import Note from '../../models/Note';
+import Setting from '../../models/Setting';
 
 export default class SearchEngineUtils {
-	static async notesForQuery(query: string, options: any = null) {
+	static async notesForQuery(query: string, options: any = null, searchEngine: any = null) {
```
Comment (laurent22) [inline packages/lib/services/searchengine/SearchEngineUtils.ts:6]: searchEngine should be of type `SearchEngine`

File: packages/lib/services/searchengine/SearchEngineUtils.ts [hunk 2]
```diff
@@ -42,16 +48,22 @@ export default class SearchEngineUtils {
 			if (idWasAutoAdded) delete sortedNotes[idx].id;
 		}
 
+		// Filter completed todos
+		const showCompletedTodos = Setting.value('showCompletedTodos');
```
Comment (laurent22) [inline packages/lib/services/searchengine/SearchEngineUtils.ts:52]: It's not necessary to create this intermediate variable

File: packages/lib/services/searchengine/SearchEngineUtils.ts [hunk 3]
```diff
@@ -42,16 +48,22 @@ export default class SearchEngineUtils {
...
+		let filterdNotes = [...sortedNotes];
```
Comment (laurent22) [inline packages/lib/services/searchengine/SearchEngineUtils.ts:53]: filterdNotes => filteredNotes

**Review by laurent22 — COMMENTED** [review]
That looks good, thanks for the fix and for adding the tests. I've just left a few minor comments.

**Comment by JackGruber** [comment]
I have made the adjustments

**Comment by laurent22** [comment]
All good now, thanks @JackGruber!

**Pull request merged**